### DISCUSSION TITLE
Graph query returning empty

### DISCUSSION THREAD

**Discussion by vasanthrj** [JanusGraph/janusgraph#3048] [Q&A]
Hi,
1.  using both indexed field and unique field
            g.V().has("idx1", "value").has("idx2Unique", "somevalue"); = returning empty results.
2. using only unique field
            g.V().has("idx2Unique", "somevalue"); =  returning vertex and it has "idx" property and value.
3. Try with indexed field and unique field with different values.
     g.V().has("idx1", "random").has("idx2Unique", "random11"); = returning a vertex

Do i need to reindex "idx1"? How to reindex only one vertex?
Please help. Stuck in production.

**Comment by li-boxuan** [comment]
To reindex one vertex, you could just update that vertex using the old property value.

**Reply by vasanthrj** [reply]
Thanks @li-boxuan Can share why this is happening? How to avoid? 
- Need to find out missing item in index.. Is it right way to get all items in a index `graph.query().has('idx1')`
